src/lib/mina.ts

```diff
@@ -773,3 +805,143 @@ function defaultNetworkState(): NetworkValue {
     nextEpochData: cloneCircuitValue(epochData),
   };
 }
+
+async function verifyAccountUpdate(
+  account: LedgerAccount,
+  accountUpdate: AccountUpdate,
+  transactionCommitments: {
+    commitment: Field;
+    fullCommitment: Field;
+  },
+  proofsEnabled: boolean
+): Promise<void> {
+  let perm = account.permissions;
+
+  let { commitment, fullCommitment } = transactionCommitments;
+
+  // we are essentially only checking if the update is empty or an actual update
+  function includesChange(val: any): boolean {
+    if (val?.constructor === Array) {
```

Comment:
I think if you use `Array.isArray(val)` as a test, TS knows that it's an array afterwards